### PR TITLE
Jk game 1

### PR DESCRIPTION
**Pull request by JonKernPA**
Oh, I forgot to commit this last night (when I think it was still Day 4).

### CODE REVIEW

File: 2_adventures/001/jonkernpa/04_day/code_walker.md [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+
+    @db ||= connection.db(database)
+    @grid ||= Mongo::Grid.new(db)
+    @authenticated ||= db.authenticate(username, password)
+
+## Hating on long constructor arg lists?
+
+Pass a hash!
+
+        def initialize(opts={})
```
Comment (kotp) [inline 2_adventures/001/jonkernpa/04_day/code_walker.md:24]: For more current versions of Ruby this is no longer necessary.  But I think the question then becomes "What exactly can I pass in that are named?"

Comment (JonKernPA) [inline 2_adventures/001/jonkernpa/04_day/code_walker.md:24]: Oh? Why is that? (I did a bit of googling but could not find where memoization is no longer needed in Ruby.)
Oh, wait. Did you mean the hash optional arg?

Comment (kotp) [inline 2_adventures/001/jonkernpa/04_day/code_walker.md:24]: Yes.  Inline comments like this are meant to comment on the line above them, as that is what appears in pull requests etc.  Some people comment on the top, which is strange to me.  So my comment was simply about line 24.

File: 2_adventures/001/jonkernpa/04_day/code_walker.md [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+          @host            = opts[:host]
+          @hosts           = opts[:hosts]
+          @connection_opts = opts[:connection_opts] || {}
+          @port            = opts[:port]
+          @database        = opts[:database] || 'dragonfly'
+          @username        = opts[:username]
+          @password        = opts[:password]
+          @connection      = opts[:connection]
+          @db              = opts[:db]
```
Comment (kotp) [inline 2_adventures/001/jonkernpa/04_day/code_walker.md:33]: This is a lot of work compared to how you can do it now.

**Comment by patmaddox** [comment]
Go you!

File: 2_adventures/001/jonkernpa/04_day/code_walker.md (hunk 2 above)
Comment (JonKernPA) [inline 2_adventures/001/jonkernpa/04_day/code_walker.md:33]: @kotp Do you mean something like this?

```
  def initialize(h)
    h.each { |key, value| send("#{key}=", value) }
  end
```

Ignoring the two fields that are not simply being set equal to the key.
Or are you talking about keyword arguments (Ruby 2.0) style?

Comment (kotp) [inline 2_adventures/001/jonkernpa/04_day/code_walker.md:33]: I don't know if I would call them keyword arguments, but named arguments Ruby 2.0 style, yes.

Keywords in Ruby (there are about 42 of them) and I think that is different from what I am thinking about.

Comment (JonKernPA) [inline 2_adventures/001/jonkernpa/04_day/code_walker.md:33]: I got that term from here [Keyword Arguments in Ruby 2.0](http://chriszetter.com/blog/2012/11/02/keyword-arguments-in-ruby-2-dot-0/).

Comment (kotp) [inline 2_adventures/001/jonkernpa/04_day/code_walker.md:33]: Yeah, I think that is perhaps a language issue.  key word arguments are different than keyword and its argument.  The assumed name due to the key being used as the word and the hash value being used as the assigned value.  Unfortunate, and I noticed it the first time I saw it and shook my head.  Naming is hard, descriptions can be harder.

**Pull request merged**